Elements list: name filtering fails after searching for an element name that does not exist

Hi,

Considered very critical:

### Steps to reproduce:
1. Visit any website.
2. Open elements list (NVDA+F7).
3. Search for an existing link.
4. Then search for a link that you know does not exist via filter field.
5. Press Home to move to the start of the text, and press Delete key.
6. Search for another link via filter field.

### Actual behavior:
Errors are thrown and the tree view is not populated.

### Expected behavior:
No problems.

### System configuration:

#### NVDA Installed/portable/running from source:
Installed

#### NVDA version:
Alpha.16126

#### Windows version:
Windows 10 Version 1803

#### Name and version of other software in use when reproducing the issue:
N/A

#### Other information about your system:

### Other questions:

#### Does the issue still occur after restarting your PC?
Yes

#### Have you tried any other versions of NVDA?
Yes - 2018.2.1.

### Traceback:
ERROR - unhandled exception (09:57:08.263):
Traceback (most recent call last):
  File "browseMode.pyc", line 1025, in onFilterEditTextChange
  File "browseMode.pyc", line 890, in filter
wxAssertionError: C++ assertion "item.IsOk()" failed at ..\..\src\msw\treectrl.cpp(1032) in wxTreeCtrl::GetItemParam(): invalid tree item

I'm thinking that this may have to do with an invalid character or tree view items being deleted.

Priority shall be set to P1 as this is a possible regression with wxPython 4 (2018.3) and due to massive impact this bug will have (cases were reported in the wild).

Thanks.

--- a/source/browseMode.py
+++ b/source/browseMode.py
@@ -887,7 +887,11 @@ class ElementsListDialog(wx.Dialog):
 	def filter(self, filterText, newElementType=False):
 		# If this is a new element type, use the element nearest the cursor.
 		# Otherwise, use the currently selected element.
-		defaultElement = self._initialElement if newElementType else self.tree.GetItemData(self.tree.GetSelection())
+		# #8753: wxPython 4 returns "invalid tree item" when the tree view is empty, so use initial element if appropriate.
+		try:
+			defaultElement = self._initialElement if newElementType else self.tree.GetItemData(self.tree.GetSelection())
+		except:
+			defaultElement = self._initialElement
 		# Clear the tree.
 		self.tree.DeleteChildren(self.treeRoot)
 

